### DISCUSSION TITLE
JSON set command

### DISCUSSION THREAD

**Discussion by novalic** [dsoftwareinc/fakeredis-py#329] [Q&A] [answered]
Hello!
I'm trying to use fakeredis to test my app.
I'm using RedisJson and I have a command that looks like this:

```
client.json().set(key, Path.root_path(), value, nx=True)
```

This is failing, giving me the following error:

`redis.exceptions.ResponseError: unknown command `json.set`, with args beginning with:`

I'm running an image of redis-stack with docker compose and testing the command and it works fine.

Anyone could help me figure out what's going on?
Thanks!

**Comment by cunla** [comment, marked as answer]
- How do you initialize client in this context?

- when installing fakeredis, did you use the json extra? 
pip install fakeredis[json]

**Comment by novalic** [comment]
I have a function to get the Redis client:
```
def get_client(...):
    return redis.Redis(...)
```

that i'm mocking in my test with a pytest fixture:
```
    @pytest.fixture
    def redis_mock(self):
        with patch('path to the file where i want the mock.get_client') as mock:
            mock.return_value = fakeredis.FakeRedis()
            yield mock
```
the mock is well initialized:
```
ipdb> client
<fakeredis._connection.FakeRedis(<redis.connection.ConnectionPool(<fakeredis._connection.FakeConnection(server=<fakeredis._server.FakeServer object at 0xffff789dd310>,db=0)>)>)>
```
am i missing some configuration?

**Reply by novalic** [reply]
forgot to add, yes, i installed fakeredis with the json extra.
i'm using poetry, pyproject.toml:
`fakeredis = {extras = ["json"], version = "^2.24.1"}`

**Comment by cunla** [comment]
Interesting.. ok, let's dig deeper and I'll try to replicate:
- what redis-py version? python version?
- can you run the following:
  ```python
  from fakeredis._commands import SUPPORTED_COMMANDS
  print(SUPPORTED_COMMANDS.keys()) # json.set should be there.
  ```
  and
  ```python
    from jsonpath_ng.ext import parse
    from redis.commands.json.path import Path
    from fakeredis.stack._json_mixin import JSONCommandsMixin, JSONObject
  ```

If these don't pass, then it is likely that `jsonpath_ng` is not installed somehow (despite the fact you added json extra)?
If these pass and you still do not have json commands support... I don't know, I might ask you to have a share screen session with me.

**Reply by novalic** [reply]
with the first piece of code i've realized that the json extra was not installed. not sure what happened, but i started from scratch and that was the issue... thanks man

**Comment by cunla** [comment]
Also, you are getting `redis.exceptions.ResponseError: unknown command json.set, with args beginning with:` How does it continues? what are key/value in your call? `client.json().set(key, Path.root_path(), value, nx=True)`

**Reply by novalic** [reply]
this was the only line i had as output
